parso/python/prefix.py

```diff
@@ -40,6 +40,14 @@ def __repr__(self):
             self.start_pos
         )
 
+    def search_ancestor(self, *node_types):
+        node = self.parent
+        while node is not None:
+            if node.type in node_types:
+                return node
+            node = node.parent
+        return None
```

Comment:
Interesting find! I would have forgotten about this. IMO it's not necessary here and we can probably remove this function. What do you think?

Reply:
[This place](https://github.com/davidhalter/parso/blob/f2b1ff942933a43c34d93e3359e7115d20cc2ee3/parso/python/pep8.py#L731) and [this place](https://github.com/davidhalter/parso/blob/f2b1ff942933a43c34d93e3359e7115d20cc2ee3/parso/python/pep8.py#L127) are calling the old `parso.tree.search_ancestor` function on `PrefixPart` objects. Since we now move the `search_ancestor` function to be a method of `NodeOrLeaf`, and `PrefixPart` is not a subclass of `NodeOrLeaf`, we need to manually add the `search_ancestor` method to `PrefixPart`. Otherwise, I think making `PrefixPart` a subclass of `NodeOrLeaf` may cause more problems.